### PR TITLE
Refactor challenge.html

### PR DESCRIPTION
**Pull request by amickan**

### CODE REVIEW

File: app/grandchallenge/core/templates/base.html [hunk 1]
```diff
@@ -66,11 +66,14 @@
...
 
             {% block messages %}
+                {% include 'challenges/challenge_banner.html' %}
                 {% include "grandchallenge/partials/messages.html" %}
             {% endblock %}
 
             {% block outer_content %}
-                {% block topbar %}{% endblock %}
+                {% block topbar %}
+                    {% include 'challenges/challenge_topbar.html' with challenge=challenge %}
```
Comment (jmsmkn) [inline app/grandchallenge/core/templates/base.html:75]: `with challenge=challenge` should be unnecessary, right? Why is it needed here and not for the banner partial?

File: app/grandchallenge/core/templates/base.html [hunk 2]
```diff
@@ -101,6 +104,9 @@
 {% block script %}
     {% include "grandchallenge/partials/script.html" %}
     {% include "timezones/partials/detect_timezone.html" %}
+    {% if challenge %}
+        <script src="{% static 'js/challenges/dropdown.js' %}" type="module"></script>
```
Comment (jmsmkn) [inline app/grandchallenge/core/templates/base.html:108]: This script line can move to app/grandchallenge/challenges/templates/challenges/challenge_topbar.html, as long as it's deferred with `<script src="{% static 'js/challenges/dropdown.js' %}" type="module" defer></script>`

File: app/grandchallenge/core/templates/base.html (hunk 1 above)
Comment (amickan) [reply app/grandchallenge/core/templates/base.html:75]: oh yes, that can go indeed!

File: app/grandchallenge/challenges/templates/challenges/challenge_topbar.html [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                       role="button">
+                            Join
+                    </a>
+                </div>
+            {% endif %}
+        </div>
+    </div>
+{% endif %}
+
+<script src="{% static 'js/challenges/dropdown.js' %}" type="module" defer></script>
```
Comment (jmsmkn) [inline app/grandchallenge/challenges/templates/challenges/challenge_topbar.html:93]:
```suggestion
    <script src="{% static 'js/challenges/dropdown.js' %}" type="module" defer></script>
{% endif %}

```

**Review by jmsmkn — APPROVED** [review]
This looks good to me. I think there will be a couple of changes but these are not vital:
- The title block is not handled so some page titles will change, this needs to be revisited as our strategy around page titles is all over the place
- The breadcrumbs will be missing for temppage, but I'm unsure what that thing is for, needs to be investigated

**Pull request merged**